crates/bevy_math/src/common_traits.rs

```diff
@@ -0,0 +1,177 @@
+use glam::{Quat, Vec2, Vec3, Vec3A, Vec4};
+use std::fmt::Debug;
+use std::ops::{Add, Div, Mul, Sub};
+
+/// A type that supports the mathematical operations of a vector space, irrespective of dimension.
+/// In particular, this means that the implementing type supports:
+/// - Scalar multiplication and division on the right by elements of `f32`
+/// - Addition and subtraction
+///
+/// Within the limitations of floating point arithmetic, all the following are required to hold:
+/// - (Associativity of addition) For all `u, v, w: Self`, `(u + v) + w == u + (v + w)`.
+/// - (Commutativity of addition) For all `u, v: Self`, `u + v == v + u`.
+/// - (Additive identity) For all `v: Self`, `v + Self::ZERO == v`.
+/// - (Additive inverse) For all `v: Self`, `v - v == v + v * (-1) == Self::ZERO`.
+/// - (Compatibility of multiplication) For all `a, b: f32`, `v: Self`, `v * (a * b) == (v * a) * b`.
```

Comment:
```suggestion
/// - (Compatibility of scalar multiplication) For all `a, b: f32`, `v: Self`, `v * (a * b) == (v * a) * b`.
```
Worth calling this explicitly scalar multiplication to allow for possible future definition of vector multiplication in another trait.

Reply:
I wrote "multiplication" because really it's "compatibility of scalar multiplication with field multiplication", which was too many words. I'm also not sure how confusing it would be in the future anyway, since vector multiplication is almost always called something else anyway (e.g. inner product, dot product, etc.).

Reply:
Honestly, that's fair. Until this trait is generic over the scalar field (probably never) I think that's specific enough language already.